packages/graphiql/src/components/HeaderEditor.tsx

```diff
@@ -71,13 +71,14 @@ export class HeaderEditor extends React.Component<HeaderEditorProps> {
     require('codemirror/addon/search/searchcursor');
     require('codemirror/addon/search/jump-to-line');
     require('codemirror/addon/dialog/dialog');
+    require('codemirror/mode/javascript/javascript');
     require('codemirror/keymap/sublime');
 
     const editor = (this.editor = this.CodeMirror(this._node, {
       value: this.props.value || '',
       lineNumbers: true,
       tabSize: 2,
-      mode: 'graphql-headers',
+      mode: { name: 'javascript', json: true },
```

Comment:
@acao This was done because the variable editor was set to `graphql-variables`, is that an actual mode or should we change that as well?

https://github.com/graphql/graphiql/blob/1124834c0751fbd560c7909b498c605fb5d0f65d/packages/graphiql/src/components/VariableEditor.tsx#L88

Reply:
`graphql-variables` is an actual mode in the `codemirror-graphql` package in this repo, as you can see in the `require()` statements
https://github.com/graphql/graphiql/blob/1124834c0751fbd560c7909b498c605fb5d0f65d/packages/graphiql/src/components/VariableEditor.tsx#L82

Reply:
codemirror is not the simplest to wrap your head around, so I can totally relate with this choice! haha